### PR TITLE
3884 -  Bulk Download -> Fra years : show only fra years (not odp year ) enhancement

### LINKED ISSUE #3884: Bulk Download -> Fra years : show only fra years (not odp year )
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by sorja**
fixes issues with headers

### CODE REVIEW

File: src/server/controller/cycleData/getBulkDownload/getContent.ts [hunk 1]
```diff
@@ -3,32 +3,37 @@ import { RecordAssessmentDatas } from 'meta/data'
...
-    countries,
-    tableNames,
-  })
+  const years = intervals
+    ? ['1990-2000', '2000-2010', '2010-2015', '2015-2020']
+    : Array.from({ length: 18 }, (_, i) => String(2000 + i))
+
+  if (cycle.name === '2025') {
+    years.push(...(intervals ? ['2020-2025'] : Array.from({ length: 7 }, (_, i) => String(2017 + i))))
+  }
```
Comment (minotogna) [inline src/server/controller/cycleData/getBulkDownload/getContent.ts]: Don't we have constants somewhere ?

File: src/server/controller/cycleData/getBulkDownload/getFRAYearsData.ts [hunk 2]
```diff
@@ -25,20 +24,8 @@ export const getFraYearsData = async (props: Props) => {
...
-  // Unique years
-  const years = getYears({
-    data,
-    countries,
-    tableNames,
-  })
-    .filter((x) => Number.isInteger(+x))
-    .sort()
+  const years = ['1990', '2000', '2010', '2015', '2020']
+  if (cycle.name === '2025') years.push('2025')
```
Comment (minotogna) [inline src/server/controller/cycleData/getBulkDownload/getFRAYearsData.ts]: And these ...

File: src/server/controller/cycleData/getBulkDownload/getContent.ts (hunk 1 above)
Comment (sorja) [reply src/server/controller/cycleData/getBulkDownload/getContent.ts]: I didn't find, should we introduce `assessment/years.ts`?

File: src/server/controller/cycleData/getBulkDownload/getFRAYearsData.ts (hunk 2 above)
Comment (sorja) [reply src/server/controller/cycleData/getBulkDownload/getFRAYearsData.ts]: [b2a7882](https://github.com/openforis/fra-platform/pull/3892/commits/b2a788206f15cad52a657a399ff65c8b5d0543ce)

File: src/server/controller/cycleData/getBulkDownload/getContent.ts (hunk 1 above)
Comment (sorja) [reply src/server/controller/cycleData/getBulkDownload/getContent.ts]: [b2a7882](https://github.com/openforis/fra-platform/pull/3892/commits/b2a788206f15cad52a657a399ff65c8b5d0543ce)

File: src/meta/assessment/years.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+const fraYears = (cycle: Cycle) => {
+  const years = Array.from({ length: 18 }, (_, i) => String(2000 + i))
+  if (cycle.name === '2025') {
+    years.push(...Array.from({ length: 7 }, (_, i) => String(2017 + i)))
+  }
+  return years
+}
+
+const intervals = (cycle: { name: string }) => {
```
Comment (minotogna) [inline src/meta/assessment/years.ts]: Can this be cycle as well ? Or Pick<Cycle ?

File: src/meta/assessment/years.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  const years = ['1990-2000', '2000-2010', '2010-2015', '2015-2020']
+  if (cycle.name === '2025') {
+    years.push('2020-2025')
+  }
+  return years
+}
+
+export const Years = {
```
Comment (minotogna) [inline src/meta/assessment/years.ts:19]: This is good 😊

File: src/meta/assessment/years.ts (hunk 3 above)
Comment (sorja) [reply src/meta/assessment/years.ts]: typo, fixed !

**Review by minotogna — APPROVED** [review]

**Pull request merged**